contracts/BandOracle.cdc

```diff
@@ -216,21 +229,21 @@ pub contract BandOracle {
 
     /// The resource that allows the maintainer account to charge a fee for the use of the oracle.
     ///
-    pub resource FeeCollector {
+    access(all) resource FeeCollector {
 
         /// Sets the fee in Flow tokens for the oracle use.
         ///
         /// @param fee: The amount of Flow tokens.
         ///
-        pub fun setFee (fee: UFix64) {
+        access(all) fun setFee (fee: UFix64) {
             BandOracle.setFee(fee: fee)
         }
 
         /// Extracts the fees from the contract's vault.
         /// 
         /// @return A vault containing the funds obtained for the oracle use.
         ///
-        pub fun collectFees (): @FungibleToken.Vault {
+        access(all) fun collectFees (): @{FungibleToken.Vault} {
             return <- BandOracle.collectFees()
         }
```

Comment:
Not a blocker but this would leverage standard events better - easier to transfer FT.Withdraw & .Deposit. Ideally we'd also include an event (would need to be defined)

```suggestion
        access(all) fun collectFees (receiver: &{FungibleToken.Receiver}): @{FungibleToken.Vault} {
            let fees <- BandOracle.collectFees()
            emit FeesCollected(amount: fees.balance, to: receiver.owner?.address, collectorUUID: self.uuid, collectorAddress: self.owner!.address)
            receiver.deposit(from: <-fees)
        }
```